editor/components/inserter/index.js

```diff
@@ -36,8 +41,12 @@ class Inserter extends wp.element.Component {
 					icon="insert"
 					label={ wp.i18n.__( 'Insert block' ) }
 					onClick={ this.toggle }
-					className="editor-inserter__toggle" />
-				{ opened && <InserterMenu position={ position } onSelect={ this.close } /> }
+					className="editor-inserter__toggle"
+					aria-haspopup="true"
+					id="inserter-toggle"
+					buttonRef={ ( node ) => this.nodes.toggle = node }
```

Comment:
> Do we really want to refocus the inserter button if we click outside?

The two will be independent, so if you click outside no focus, unless if you hit the toggle. If you close the menu purely with the keyboard, it will focus the toggle button for better accessibility.